src/components/App.js

```diff
@@ -16,7 +18,12 @@ const App = () => {
   return (
     <main>
       <NavBar />
-      <BooksContainer books={books}/>
+      <Routes>
+        <Route path='/' element={<BooksContainer books={books} />} />
+        <Route path='/details' element={<BookDetails books={books}/>}>
+          <Route path='/details/:id' element={<BookDetails />}/>
```

Comment:
I think you could also do `path='/:id` here since it's nested in the `details` route